### ISSUE TITLE
tomviz crashes when running out of memory

### ISSUE THREAD

**Issue by yijiang1** [OpenChemistry/tomviz#822]

**Comment by cryos** [comment]
Yeah, this is a tough one to get right, and there are a number of places this can cause crashes. What operating system are you seeing these crashes on? It would be good to improve reliability here, on Linux Tomviz typically gets very slow due to excessive swapping, and may be killed by the kernel after some time due to excessive memory use/swapping but it rarely crashes.

**Comment by yijiang1** [comment]
It crashes on Mac OSX.
